firebase-database/src/main/java/com/google/firebase/database/core/CompoundWrite.java

```diff
@@ -31,7 +31,7 @@
  * is only allowed to be one write modifying that path. Any write to an existing path or shadowing
  * an existing path will modify that existing write to reflect the write added.
  */
-public class CompoundWrite implements Iterable<Map.Entry<Path, Node>> {
+public final class CompoundWrite implements Iterable<Map.Entry<Path, Node>> {
```

Comment:
`equals` cannot check class equality of the parameter (isa CompoundWrite); they should use an instanceOf check or make the class final.